### DISCUSSION TITLE
List of books in series that are not in the library

### DISCUSSION THREAD

**Discussion by nosTa1337** [joonaspaakko/audible-library-extractor#100] [Ideas]
Hi,

it would be nice to have a feature where you can click "show series books not in my library" or something similar, where you have a complete overview over all the books without clicking on each one. Or show on indicator on the book that there are still books available not listened to from the series. Would help a lot. Before your addon I went to the app, scrolled through my library, selected the book I am awaiting a new release, clicking the series and so on and so on. As you mostly don't know when it comes available it could be very useful.

If I find some time I may do a pull request, but right now I have not much time to spare.

Best regards,
Philipp

**Comment by joonaspaakko** [comment]
I think that already exists in the form of [`my book in the series list`](https://joonaspaakko.gitbook.io/audible-library-extractor/gallery/my-books-in-the-series-list) on the book details view. That documentation page gives you a good idea of what information this list gives you and what its functionality is, but [here's an image to show a bit better what it looks like with all the missing books mixed in and how the filtering works](https://user-images.githubusercontent.com/1164476/176848110-ce00564d-ca91-475a-8cb6-8fc3b0245f12.png).

Admittedly this compact series list is a little disjointed from the series sub-page view, which I think is fine. But it's pretty wonky if you prefer to look at the series sub-page. It feels a little weird that to learn more about the series you're already looking at in the series sub-page, you have to click open the book details for any book in the series.

**TL;DR showing "missing books" in the series sub-page is in my todo list (for the next version)**

>
> Originally (even before the `my book in the series list` existed) I deemed it too much of a hassle to show "missing books" in the series sub-page, because I would need to extract more and more data. Nowadays the only missing piece in terms of data is the cover image of every single book in a series, and since some other data for those books is already getting fetched, what's one image URL on top of that. **So yea, since it was suggested in [another post](https://github.com/joonaspaakko/audible-library-extractor/discussions/95), I've decided to add that**, as I talked about [in here](https://github.com/joonaspaakko/audible-library-extractor/discussions/96#discussioncomment-2900891).
> 
> - I do have plans to "copy over" most of the functionality from the `my book in the series list`, like when a missing book is clicked, it will not be opening book details, but it will instead open the book in Audible. Opening book details is too much to ask. It would required extracting data from the store page for each book in a series and that would not only take forever, but I would guess ut would easily triple the filesize. 
>
> - I'm not sure if I'll be adding a new color dot indicator or any kind of indicator for books that you've stared (like in the `my book in the series list`). I might, but for now I'm feeling like it would add more unnecessary clutter on the covers, so I'm thinking I won't. I will try and see if it would make sense to add it. 
>
> - Another thing I won't be adding to the series sub-page is how `my book in the series list` [shows all the series the book belongs to (image)](https://user-images.githubusercontent.com/1164476/176850183-7cd7b320-6151-4c15-abdb-03e34201d2a9.png), which just isn't something that can be replicated on the sub-page for obvious reasons. You'll just have to open the book details and see it from there.
>
> - And none of the icon's or their functionality from the `my book in the series list` won't be added to the series sub-page, because it already has filters. That said, I'm definitely adding some kinda mechanism to show and hide missing books, but this might be a toggle button below the search bar or something... or just a filter, just like anything else.

**Reply by nosTa1337** [reply]
Thanks for the detailed answer. If I can lend you a hand somewhere let me know.

**Reply by joonaspaakko** [reply]
Before I forget, ~~I said that I can't copy the feature to series sub page that lists all series from my books in the series list...~~ But it occurred to me that I could get closer to that by adding links to "related series" (probably above or below search in series sub pages).

So basically: loop through every book in series sub page, find every single series listen under each book and generate (unique) links to those sub pages.

It could perhaps sort the links by total number of books in each series and show the number of books in each link. Sort of like in the series page.

> However... It could only list series that at least one book in the library is also in, because this data lives in the store page and I won't be fetching that for every single book in the series (for as long as is possible).
